all-together.ipynb

```diff
@@ -0,0 +1,775 @@
+{
```

Comment:
<p>Instead of iterating over all pairs in such manner, you can do one of the following things (worst to best)</p><ol><li>Remove the <code>i &lt; j</code> check by iterating over <code>dirty_dataset[i + 1:]</code></li><li>Iterate over all pairs with <code>import itertools</code> and <code>for d1, d2 in itertools.combinataions(dirty_dataset, 2):</code> (check the itertools module, it has some very helpful functions)</li><li>Group all the data points by intent, then concatenate items inside each list. For example, you can do the following (naming is not perfect and <code>data_by_intent</code> should be renamed):</li></ol><pre class="ql-syntax" spellcheck="false">from collections import defaultdict

data_by_intent = defaultdict(list)
for d in dirty_dataset:
   data_by_intent[d['intent']].append(d)
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/JetBrains-Research/metrics-evaluation/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>